### PR TITLE
Dkobylia/clic pflow

### PR DESCRIPTION
**Pull request by samvanstroud**
reopening #33 using a branch in this repo.

---------

Adds CLIC MPflow example. 
It's a draft version, aimed to get the same performance as with the old codebase fast.

Introduce the following changes:

Loss functions:
- Add kldiv, CE losses
- Change cost and loss calculation to the old repo format

Matcher:
- Update lap1015 code to version in [dev branch](https://github.com/sguthe/lap-solver/tree/dev) -- not sure if it's needed, but that's what I had
- Match only valid part of the cost matrix
	- Works faster with scipy solver, test in `test_match equivalence`
	- Speed with lap1015 is not affected
	
Tasks:
- Add option to skip task in intermediate levels and keep only in final level
- Add `ObjectClassificationTask` for multi-class object classification -- probably can be merged with `ObjectValidTask`
- Add `IncidenceRegressionTask` for incidence prediction, very similar to regular mask prediction task, but with softmax inside
- Add `IncidenceBasedRegressionTask` that uses the incidence matrix to calculate proxy features and also importance of node features
	- It allows to split charge and neutral predictions, since it makes less sense to predict energy for charged particles
	- Loss calculation is performed only on valid particles (where target is valid and prediction is valid)
	
CLIC Pflow:
- Add dataset implementation
- Add custom `predictionwriter`, because we use ROOT in performance evaluation
- `Performance` folder is inherited from Nilotpal's HGPflow repo

Misc:
- Changes in `transformer` and `wrapper` are unnesessary, should be reverted and updated wrt main

### CODE REVIEW

File: src/hepattn/models/maskformer.py [hunk 1]
```diff
@@ -175,6 +190,14 @@ def forward(self, inputs: dict[str, Tensor]) -> dict[str, Tensor]:
         for task in self.tasks:
             outputs["final"][task.name] = task(x)
 
+            # Need this for incidence-based regression task
+            if task.name == "incidence":
+                # Assume that the incidence task has only one output
+                x["incidence"] = outputs["final"][task.name][task.outputs[0]].detach()
+            if task.name == "classification":
+                # Assume that the classification task has only one output
+                x["class_probs"] = outputs["final"][task.name][task.outputs[0]].detach()
```
[the remark this answers is not in the document]
Comment (samvanstroud) [reply src/hepattn/models/maskformer.py:216]: This one still needs addressing @sunnyw22
